cli/internal/config/config.go

```diff
@@ -182,9 +201,59 @@ func ParseAndValidate(args []string, ui cli.Ui, turboVersion string) (c *Config,
 			Workers: runtime.NumCPU() + 2,
 			Dir:     filepath.Join("node_modules", ".cache", "turbo"),
 		},
+		RootPackageJSON: rootPackageJSON,
+		TurboConfigJSON: turboConfigJson,
+		Cwd:             cwd,
 	}
 
 	c.ApiClient.SetToken(partialConfig.Token)
 
 	return c, nil
 }
+
+func ReadTurboConfig(rootPath string, rootPackageJSON *fs.PackageJSON) (*fs.TurboConfigJSON, error) {
+	// If turbo.json exists, we use that
+	// If pkg.Turbo exists, we warn about running the migration
+	// Use pkg.Turbo if turbo.json doesn't exist
+	// If neither exists, it's a fatal error
+	turboJSONPath := filepath.Join(rootPath, "turbo.json")
+
+	if !fs.FileExists(turboJSONPath) {
+		if rootPackageJSON.LegacyTurboConfig == nil {
+			// TODO: suggestion on how to create one
+			return nil, fmt.Errorf("Could not find turbo.json. Follow directions at https://turborepo.org/docs/getting-started to create one")
+		} else {
+			log.Println("[WARNING] Turbo configuration now lives in \"turbo.json\". Migrate to turbo.json by running \"npx @turbo/codemod create-turbo-config\"")
+			return rootPackageJSON.LegacyTurboConfig, nil
+		}
+	} else {
+		turbo, err := fs.ReadTurboConfigJSON(turboJSONPath)
+		if err != nil {
+			return nil, fmt.Errorf("turbo.json: %w", err)
+		}
+		if rootPackageJSON.LegacyTurboConfig != nil {
+			log.Println("[WARNING] Ignoring legacy \"turbo\" key in package.json, using turbo.json instead. Consider deleting the \"turbo\" key from package.json")
+			rootPackageJSON.LegacyTurboConfig = nil
+		}
+		return turbo, nil
+	}
+}
+
+// Selects the current current working directory from OS
+// and overrides with the `--cwd=` input argument
+func selectCwd(inputArgs *[]string) (string, error) {
```

Comment:
go style nit: slices are essentially already pointers, you don't need to pass pointers to them.